port/osx/omrosdump.c

```diff
@@ -442,6 +442,14 @@ omrdump_create(struct OMRPortLibrary *portLibrary, char *filename, char *dumpTyp
 			mach_error("failed get special port:\n", kr);
 			return kr;
 		}
+		/* Move to specified folder before dumping */
+		if (NULL != lastSep) {
+			lastSep[1] = '\0';
```

Comment:
This should be
```
    lastSep[0] = '\0';
```
and the separator needs to be restored to fulfill the contract of this API (providing the name of the core file actually created in `filename`).

Reply:
On further reflection, perhaps it is correct as is, the filename might be `/core.NNN` in which case we want to `chdir("/")`: I think a comment would help here.

Reply:
I copied the line from https://github.com/eclipse/omr/blob/9494694dc394aa5dce88fb4d101a9a7ed17ca38b/port/unix/omrosdump.c#L176
which also preserves the separator.

Added a comment explaining why we keep the separator.

Reply:
Thanks, @mikezhang1234567890.